sqlglot/parser.py

```diff
@@ -6671,12 +6677,15 @@ def _parse_credentials(self) -> t.Optional[exp.Credentials]:
         expr = self.expression(exp.Credentials)
 
         if self._match_text_seq("STORAGE_INTEGRATION", advance=False):
-            expr.set("storage", self._parse_conjunction())
+            expr.set("storage", self._parse_property(parse_as_var=True))
         if self._match_text_seq("CREDENTIALS"):
-            # Snowflake supports CREDENTIALS = (...), while Redshift CREDENTIALS <string>
-            creds = (
-                self._parse_wrapped_options() if self._match(TokenType.EQ) else self._parse_field()
-            )
+            creds: t.Optional[exp.Expression] | t.List[t.Optional[exp.Expression]] = None
+            if self._match(TokenType.EQ):
+                # Snowflake case: CREDENTIALS = (...) . Note that empty parens are allowed too, thus the var fallback
```

Comment:
Outdated comment

Reply:
```suggestion
```

Reply:
Thanks for the catch. I added a `quote_identifiers` test to check this by proxy; If anything wasn't parsed properly it wouldn't (or would) be quoted, so this kills 2 birds with one stone. Wdyt?